internal/infrastructure/kubernetes/deployment.go

```diff
@@ -35,24 +36,19 @@ const (
 	envoyHTTPPort = int32(8080)
 	// envoyHTTPSPort is the container port number of Envoy's HTTPS endpoint.
 	envoyHTTPSPort = int32(8443)
+	// envoyGatewayXdsServerHost is the address of the Xds Server within Envoy Gateway.
```

Comment:
`envoyGatewayXdsServerHost is the address` seems confusing. I suggest explaining in the docs that it's either an IP address or DNS 1123 label name. We should also mention that the default value is the name of the Envoy Gateway Kubernetes Service.

Reply:
since its set to https://github.com/envoyproxy/gateway/blob/13c1517bb07eb3c536fdd076bbd330a3521ce5ba/internal/infrastructure/kubernetes/bootstrap.yaml.tpl#L43 it must be a hostname, I'll update the comment